pulsar-client-cpp/python/src/config.cc

```diff
@@ -18,6 +18,99 @@
  */
 #include "utils.h"
 
+class LoggerWrapper: public Logger {
+    std::string _logger;
+    PyObject* _pyLogger;
+
+public:
+
+    LoggerWrapper(const std::string &logger, PyObject* pyLogger) : _logger(logger) {
+        _pyLogger = pyLogger;
+        Py_XINCREF(_pyLogger);
+    }
+
+    LoggerWrapper(const LoggerWrapper& other) {
+        _pyLogger = other._pyLogger;
+        Py_XINCREF(_pyLogger);
+    }
+
+    LoggerWrapper& operator=(const LoggerWrapper& other) {
+        _pyLogger = other._pyLogger;
+        Py_XINCREF(_pyLogger);
+        return *this;
+    }
+
+    virtual ~LoggerWrapper() {
+        Py_XDECREF(_pyLogger);
+    }
+
+    bool isEnabled(Level level) {
```

Comment:
It might be overkill to get back into Python each time to check the log level. This will be called many times (eg. for the debug statements). is the logging level supposed to be changed frequently at runtime? Could we cache it in c++ land and only check on python every 1sec or so?

Reply:
For example, we have a `RateLimiter` in the `perf` directory. We could try to reuse the same class to "limit" the number of times/s we're checking what's the current log level

Reply:
Thank you for the suggestion! It seems to me that `RateLimiter` blocks the caller and I think that it's not the intended behavior. Please correct me if I am wrong.

Taking into account that in real life you would not want to change the level at run-time very often, I propose the following options:

**Solution 1**:
* Initialize a class member `currentLoggingLevel` during the construction of the Logger, by calling the python's "logger.getEffectiveLevel()". Then never update this value. This would imply that whenever the python programmer wants to change the logging behavior he would have to recycle the Client object.

**Solution 2**:
* Add an internal counter to the logger and only forward the call to python each Nth time `Logger::isEnabled` is called.